### ISSUE TITLE
Checkpointing by step in Torchtune

### ISSUE THREAD

**Issue by 00HS** [pytorch/torchtune#2905]
Hi, I was going through the reference YAML config files and noticed that checkpoint saving is not done on a per-step basis. 
For larger datasets, it seems more practical to save checkpoints (and monitor validation loss) at certain step intervals rather than per epoch.

Is it currently possible to save checkpoints at every n steps?
If not, is there any plan to support step-based checkpointing and validation in the future?

**Comment by iqiancheng** [comment]
If you want, you can refer to [save_every_n_steps](https://github.com/pytorch/torchtune/blob/dc0441a095039dcd950c0afd916657b91c63835d/recipes/lora_finetune_distributed.py#L193)

**Comment by 00HS** [comment]
Thanks for @iqiancheng!
When I first tried it, it didn’t work, but after the update, it works perfectly!
